### ISSUE TITLE
  [Feature] [pi05] Batch multiple camera views before vision tower to avoid per-view serial forward

### ISSUE THREAD

**Issue by XueSongTap** [baidu-baige/LoongForge#16]
## Background

  In `loongforge/models/embodied/pi05/modeling_pi05.py`, `PI05Pytorch.embed_prefix()` currently processes multi-camera observations by iterating over `images/img_masks`:

  ```python
  for img, img_mask in zip(images, img_masks, strict=True):
      img_emb = self._apply_checkpoint(image_embed_func, img)
      ...
      embs.append(img_emb)
```
  Each img corresponds to one camera view, and each view independently calls:
```
  self.paligemma_with_expert.embed_image(img)
```
  which eventually runs the PaliGemma/SigLIP vision tower.

  This means multiple camera views are forwarded through the same vision tower one by one. The output tokens are then concatenated as:

  [cam0_image_tokens, cam1_image_tokens, ..., language_tokens]

  ## Problem

  The current implementation is functionally correct, but the per-camera Python loop introduces unnecessary serialization:

  - each camera view triggers a separate vision tower forward;
  - CUDA kernels are launched multiple times instead of once over a larger batch;

  This is especially relevant for multi-view embodied learning setups where observations commonly include several camera streams.

  ## Proposed Feature

  Add an optimized path that batches camera views before calling the vision tower.

  use:

  images:      list[V] of [B, C, H, W]
  stack:       [B, V, C, H, W]
  flatten:     [B * V, C, H, W]
  vision:      [B * V, N, D]
  reshape:     [B, V, N, D]
  flatten seq: [B, V * N, D]
  prefix:      concat([image_tokens, lang_emb], dim=sequence)

  The corresponding image masks can be transformed similarly:

  img_masks: [B, V]
  expand:    [B, V, N]
  reshape:   [B, V * N]

  This keeps the current token ordering unchanged:

  cam0 tokens -> cam1 tokens -> ... -> language tokens

  ## Expected Benefits

  - Reduce repeated Python-level loop overhead.
  - Reduce repeated vision tower forward dispatches.
  - Improve MFU by increasing the effective vision batch size.
  - Lower multi-camera training latency.
  - Preserve the existing model semantics, token order, attention masks, and checkpoint compatibility.
